tests/OK/767.lp

```diff
@@ -1,4 +1,11 @@
 constant symbol Set : TYPE;
 injective symbol τ : Set → TYPE;
 constant symbol B : Set;
-type λ (f : _ → _ → τ B) (g : τ B → τ B) b1 b2, f (g b1) (g b2);
+type λ (f : _ → τ B) (b: τ B), f b;
+// type λ (f : _ → _ → τ B) (g : τ B → τ B) b1 b2, f (g b1) (g b2);
+// The command above fails because the instantiation of the second placeholder
+// triggers a unification problem τ B == ?1[g b1], which fails because `g b1`
+// is not a variable.
+// "arguments are not distinct variables: g b1"
+// It worked before this PR because the arrow notation would specify that the
+// second meta-variable cannot depend on the first argument.
```

Comment:
That's annoying. This should be documented in CHANGES.

Reply:
Some idea: One way to recover this example and improve the efficiency of type inference also perhaps would be to add in the type term a new constructor Arrow for the non-dependent product. In type inference and unification, we could even use a new smart constructor mkArrow_or_Prod to convert a closed dependent product into a non-dependent product.

Reply:
I'll add an issue afteward